### PR TITLE
Painting fence problem in java

### LINKED ISSUE #7157: Painting Fence Algorithm in java , javascript
## 🚀 Feature

To Implement painting fence algorithm in java and python

### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#reporting-new-issues)?

yes

## Motivation

To Implement painting fence algorithm in java and python

## Pitch

To Implement painting fence algorithm in java and python

### PR DESCRIPTION
**Pull request by aaadddiii**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

yes

### Description

Fixes #7157 

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

issue #7157

### CODE REVIEW

**Comment by aaadddiii** [comment]
@HarshCasper  @ankitaggarwal23  can you please review my pr

File: Java/README.md [hunk 1]
```diff
@@ -142,11 +142,12 @@ _add list here_
 - [Egg Dropping Problem](dp/EggDroppingProblem.java)
 - [Travelling Salesman Problem](dp/Traveling_SalesmanDP.java)
 - [Longest Bitonic Subsequence](dp/Longest_bitonic_subsequence.java)
+- [Painting fence problem](dp/Painting_fence.java)
 
 ## Data Structures
 
 - [Binary Tree](ds/Binary_Tree.java)
-- [Level order traversal of a tree](ds/LevelOrderTraversalRecursion.java)
+- [evel order traversal of a tree](ds/LevelOrderTraversalRecursion.java)
```
Comment (ankitaggarwal23) [inline Java/README.md]: Why you update this line?

File: Java/dp/Painting_fence.java [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+second fence with the same color as the first post is k, and the number of ways 
+of painting the second post with a different color is k*(k-1) as we have k-1 colors 
+that are different from the first post's color. So the total number of ways of 
+painting 2 fences is k*(k-1) + k. Now we consider 3 posts, Then the number of ways 
+of painting the third fence same as the second fence is the number of ways of painting 
+the second fence with a different color that is k*(k-1). The number of ways of painting 
+the third fence using a different color is equal to the total number of ways of painting 
+the second fence times (k-1). Hence the total number of ways of painting 3 fences is 
+k*(k-1) + (k+k*(k-1))*(k-1). This way we can find the total number of ways of painting 
+n fences
```
Comment (ankitaggarwal23) [inline Java/dp/Painting_fence.java]: Remove this part

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]

**Comment by aaadddiii** [comment]
@ankitaggarwal23  I have made the changes can you please review my pr

**Review by ankitaggarwal23 — APPROVED** [review]

**Pull request merged**